### ISSUE TITLE
Vasprun get_potcars fails if no explicit directories in file path

### ISSUE THREAD

**Issue by bfield1** [materialsproject/pymatgen#3604]
### Python version

3.9.18

### Pymatgen version

2024.1.27

### Operating system version

MacOS 13.6.4

### Current behavior

When loading a Vasprun using a 'vasprun.xml' file in the present working directory (i.e. `Vasprun('vasprun.xml')`), it throws a FileNotFoundError when it attempts to `get_potcar`. The traceback is below.

Specifically, in `get_potcar`, it uses `os.path.split` to get the directory of the filename. But in this case, that is an empty string. The code then fails in the next line, when it attempts to evaluate `os.listdir('')`.

If I instead call `Vasprun('./vasprun.xml')` or `Vasprun('vasprun.xml', parse_potcar_file=False)` the code runs without error.

```python3
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[2], line 1
----> 1 Vasprun('vasprun.xml')

File ~/anaconda3/envs/ifermi/lib/python3.9/site-packages/pymatgen/io/vasp/outputs.py:296, in Vasprun.__init__(self, filename, ionic_step_skip, ionic_step_offset, parse_dos, parse_eigen, parse_projected_eigen, parse_potcar_file, occu_tol, separate_spins, exception_on_bad_xml)
    293         self.nionic_steps = len(self.ionic_steps)
    295     if parse_potcar_file:
--> 296         self.update_potcar_spec(parse_potcar_file)
    297         self.update_charge_from_potcar(parse_potcar_file)
    299 if self.incar.get("ALGO") not in ["CHI", "BSE"] and not self.converged and self.parameters.get("IBRION") != 0:

File ~/anaconda3/envs/ifermi/lib/python3.9/site-packages/pymatgen/io/vasp/outputs.py:1054, in Vasprun.update_potcar_spec(self, path)
   1046 def update_potcar_spec(self, path):
   1047     """
   1048     Args:
   1049         path: Path to search for POTCARs
   (...)
   1052         Potcar spec from path.
   1053     """
-> 1054     if potcar := self.get_potcars(path):
   1055         self.potcar_spec = [
   1056             {"titel": sym, "hash": ps.md5_header_hash, "summary_stats": ps._summary_stats}
   1057             for sym in self.potcar_symbols
   1058             for ps in potcar
   1059             if ps.symbol == sym.split()[1]
   1060         ]

File ~/anaconda3/envs/ifermi/lib/python3.9/site-packages/pymatgen/io/vasp/outputs.py:1010, in Vasprun.get_potcars(self, path)
   1007 else:
   1008     search_path = os.path.split(self.filename)[0] if path is True else str(path)
   1009     potcar_paths = [
-> 1010         f"{search_path}/{fn}" for fn in os.listdir(search_path) if fn.startswith("POTCAR") and ".spec" not in fn
   1011     ]
   1013 for potcar_path in potcar_paths:
   1014     try:

FileNotFoundError: [Errno 2] No such file or directory: ''
```

### Expected Behavior

`Vasprun.get_potcar` should be able to handle the case where `self.filename` is a relative path with no directories.

This could be done by adding a line `if search_path == '': search_path = None`. This will then work as expected, because `os.listdir(None)` acts as if the argument were the present working directory.

### Minimal example

In a directory containing a `vasprun.xml` file...
```Python
from pymatgen.io.vasp.outputs import Vasprun
Vasprun('vasprun.xml')
```

**Comment by esoteric-ephemera** [comment]
Thanks @bfield1, this is a duplicate of #3586, and should be fixed in #3587. Can you pip install from git and see if this issue is resolved (`pip install git+https://github.com/materialsproject/pymatgen.git`)?

**Comment by esoteric-ephemera** [comment]
No worries I'm glad to hear this is fixed!
